packages/wrangler/src/autoconfig/details.ts

```diff
@@ -147,3 +203,73 @@ export function displayAutoConfigDetails(
 
 	logger.log("");
 }
+
+export async function confirmAutoConfigDetails(
+	autoConfigDetails: AutoConfigDetails
+): Promise<AutoConfigDetails> {
+	const modifySettings = await confirm(
+		"Do you want to modify these settings?",
+		{ defaultValue: false, fallbackValue: false }
+	);
+
+	if (!modifySettings) {
+		return autoConfigDetails;
+	}
+
+	const { framework, ...detailsWithoutFramework } = autoConfigDetails;
+	const updatedAutoConfigDetails: AutoConfigDetails = structuredClone(
+		detailsWithoutFramework
+	);
+	if (framework) {
+		// The framework cannot be `structuredClone`d so we just copy it here
+		updatedAutoConfigDetails.framework = framework;
+	}
```

Comment:
Could you explain what this logic is doing?

Reply:
I agree about the strings and the fact that things are mostly one level deep now, but things might change (or not 😅 ) later down the line 🤷 

I really am ok with whatever we prefer to go with here, I also have a feeling that the logic here might be subject to change as we developer autoconfig more, so definitely I'm happy to just go with what we find more sensible for the moment and allow things to evolve as/if needed